### PR TITLE
nodePackages.musescore-downloader: use buildNpmPackage

### LINKED ISSUE #244285: buildNpmPackage fails to build pdfkit
### Describe the bug

I'm trying to package https://github.com/LibreScore/dl-librescore which directly depends on https://github.com/foliojs/pdfkit.

I get the following error:
```
Error: Git dependency node_modules/pdfkit contains install scripts, but has no lockfile, which is something that will probably break. Open an issue if you can't feasibly patch this dependency out, and we'll come up with a workaround.
If you'd like to attempt to try to use this dependency anyways, set `forceGitDeps = true`.
```

If I set `forceGitDeps = true`, then I get the following error from npm:
```
npm ERR! Your cache folder contains root-owned files, due to a bug in
npm ERR! previous versions of npm which has since been addressed.
npm ERR! 
npm ERR! To permanently fix this problem, please run:
npm ERR!   sudo chown -R 1000:100 "/nix/store/qxdmy3k4s2ysgdkahi3497jab735jxz1-dl-librescore-0.34.46-npm-deps"
```

The error messages are good enough at explaining why it's a problem, but I don't know how to fix it. I'd appreciate some help!

I cannot feasibly remove the pdfkit dependency because generating PDFs is essential functionality of this package.

### Steps To Reproduce
1. try to build the package `dl-librescore` from https://github.com/ners/nixpkgs/tree/dl-librescore, e.g. by issuing:
  ```console
  nix build github:ners/nixpkgs/dl-librescore#dl-librescore
  ```

### Expected behavior
The package builds.

cc @lilyinstarlight

### PR DESCRIPTION
**Pull request by dotlambda**
## Description of changes
see #229475 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @legendofmiracles who originally packaged it in https://github.com/NixOS/nixpkgs/pull/131382
closes https://github.com/NixOS/nixpkgs/issues/244285 (cc @ners)

### CODE REVIEW

File: pkgs/tools/audio/dl-librescore/default.nix
Comment (dotlambda) [inline pkgs/tools/audio/dl-librescore/default.nix:34]: @ners Do you want to be the maintainer?

**Review by lilyinstarlight — DISMISSED** [review]
This approach seems sound, thank you for doing this!

If you have the spare bandwidth, it might be good to include a bit of message in your upstream PR briefly explaining why this matters and asking to ensure they don't truly need an unreleased pdfkit for some weird reason

Either way, this looks like the correct approach and I appreciate you handling this :)

(Also I'll wait a bit longer on @ners or @legendofmiracles before merging this, since it would be nice for it to have a maintainer in the list before merging)

File: pkgs/tools/audio/dl-librescore/default.nix
Comment (lilyinstarlight) [inline pkgs/tools/audio/dl-librescore/default.nix:28]:
```suggestion

  nativeBuildInputs = [ python3 ] ++ lib.optionals stdenv.isDarwin [ darwin.cctools ];

```

Ah, actually this needs `python3` and `libtool` from `darwin.cctools` for the node-gyp build. Surprisingly it only needs this on aarch64-darwin, but at least `python3` should be included on all platforms if it's doing node-gyp stuff...

File: pkgs/tools/audio/dl-librescore/default.nix
Comment (lilyinstarlight) [inline pkgs/tools/audio/dl-librescore/default.nix:23]: Can you hoist this comment above the `postPatch` (so it doesn't affect the drv hash) and mark it with `TODO: can be removed [insert condition here]` so it's clear when this `postPatch` can be removed?

Thank you again for this!

Comment (dotlambda) [reply pkgs/tools/audio/dl-librescore/default.nix:23]: I don't like the former because it makes things confusing if multiple things are done in `postPatch`.

Comment (lilyinstarlight) [reply pkgs/tools/audio/dl-librescore/default.nix:23]: That's fine then, it's more a personal preference so I probably shouldn't have asked for it anyway. Feel free to leave it inside the `postPatch`, but it would be good to know when it can be removed

File: pkgs/development/node-packages/aliases.nix
Comment (WolfangAukang) [inline pkgs/development/node-packages/aliases.nix:82]:
```suggestion
  musescore-downloader = pkgs.dl-librescore; # added 2023-08-19
```

**Pull request merged**
